src/leo_s3_bucket.erl

```diff
@@ -880,20 +880,47 @@ find_bucket_by_name_1(Bucket, DB, Providers) ->
 
     Ret = lists:foldl(
             fun(Node, null) ->
-                    find_bucket_by_name_2(Bucket, DB, Node, Value);
+                    find_bucket_by_name_2(Bucket, Node, Value);
                (Node, {error, _Cause}) ->
-                    find_bucket_by_name_2(Bucket, DB, Node, Value);
+                    find_bucket_by_name_2(Bucket, Node, Value);
                (_Node, SoFar) ->
                     SoFar
             end, null, Providers),
-    Ret.
+    Value_1 = case Ret of
+                  {ok, NewValue} ->
+                      NewValue;
+                  not_found ->
+                      not_found;
+                  _ ->
+                      Value
+              end,
+    case Value_1 of
+        null ->
+            not_found;
+        not_found ->
+            not_found;
+        _ ->
```

Comment:
When Value_1 matches {error, any()} due to timeout or badrpc,
It'd be good to log with warn level for informing admins that sync with leo_manager failed.